llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -629,12 +629,16 @@ SIRegisterInfo::getMaxNumVectorRegs(const MachineFunction &MF) const {
   return std::pair(MaxNumVGPRs, MaxNumAGPRs);
 }
 
-BitVector SIRegisterInfo::getReservedRegs(const MachineFunction &MF) const {
-  BitVector Reserved(getNumRegs());
+BitVector SIRegisterInfo::getAlwaysReservedRegs() const {
+  static BitVector AlwaysReservedRegs;
+  // Already been calculated, so do not compute again.
+  if (AlwaysReservedRegs.size() == getNumRegs())
```

Comment:
Doesn't this need to be done in a thread-safe way??

Reply:
Ideally we would just generate this as a constant mask in tablegen